api/v1alpha1/gateway_types.go

```diff
@@ -335,13 +346,23 @@ type ListenerExtensionObjectReference = ConfigMapsDefaultLocalObjectReference
 
 // GatewayStatus defines the observed state of Gateway.
 type GatewayStatus struct {
+	// Addresses lists the IP addresses that have actually been
+	// bound to the Gateway. These addresses may differ from the
+	// addresses in the Spec, e.g. if the Gateway automatically
+	// assigns an address from a reserved pool.
+	//
+	// These addresses should all be of type "IPAddress".
```

Comment:
Why not a NamedAddress? Isn't that what we'd see in status for an AWS ELB?

Reply:
The intent is that this address should be something contactable. IIUC "NamedAddress" isn't a hostname, it's an abstract name to some address resource the controller holds and can only be interpreted by the controller.

Reply:
Oh right, I'd forgotten that bit. It seems like we may be missing a potential address type then? That doesn't necessarily have to be solved in this PR.

Reply:
> it's an abstract name to some address resource the controller holds and can only be interpreted by the controller.

TIL. I might have missed this before.
So, in that case, how will the address of an AWS ELB show up? It is not a NamedAddress and not an IP address.

Reply:
Not a clue. We should add some API surface to support that :)

Reply:
What is an AW ELB address? (DNS entry?)

Reply:
I'm not really familiar with it, but I've heard it rumored it is a DNS entry :)

Reply:
An [AWS ELB](https://docs.aws.amazon.com/sdk-for-go/api/service/elb/#LoadBalancerDescription) _address_ **could** refer to one of the following:

* the LB name (this is not the address, but I'm including it as there were questions about ELB in general it seemed)
* a DNSName (see [`LoadBalancerDescription`](https://docs.aws.amazon.com/sdk-for-go/api/service/elb/#LoadBalancerDescription))
* or the IP address to which the `DNSName` resolves. Please note this can also be an elastic IP instead of the automatically assigned value
* an IP address (which *can* be from an elastic IP)